### PR TITLE
Fix bug where objects and arrays marked as secret weren't being masked

### PR DESCRIPTION
**Pull request by nmaludy**
Most likely due to the re-work of secret masking here https://github.com/StackStorm/st2/pull/4140 `objects` and `arrays` that were marked with `secret: true` were not being masked. See https://github.com/StackStorm/st2/issues/4235 for details on the symptoms. 

This PR addresses this by looking for the `secret` property on `object` and `array` types. It then treats all data within the `object` or `array` as secret, returning a single masked value.

Example of the fixed version:

```
$ st2 run default.secret_objects secret_object='{"abc": "123"}' secret_array='[1, 2, 3]'
.
id: 5b44de76a814c004961c020e
status: succeeded
parameters: 
  secret_object: '********'
  secret_array: '********'
result: 
  failed: false
  return_code: 0
  stderr: ''
  stdout: ''
  succeeded: true
```

FYI, i create a unit test for this edge case so we can verify that it is not exposed in the future.

### CODE REVIEW

**Comment by nmaludy** [comment]
Closes https://github.com/StackStorm/st2/issues/4235

**Comment by nmaludy** [comment]
@Kami i was able to find and fix the issue, please let me know if you have any comments

**Comment by Kami** [comment]
@nmaludy Thanks, will have a look.

File: st2common/tests/unit/test_util_secrets.py [hunk 1]
```diff
@@ -669,3 +701,39 @@ def test_mask_secret_parameters_nested_array_with_object(self):
             ]
         }
         self.assertEqual(expected, result)
+
+    def test_mask_secret_array(self):
```
Comment (Kami) [inline st2common/tests/unit/test_util_secrets.py]: It also wouldn't hurt to add a test case for scenarios where we have an object which is marked "secret: true" at top level, but it also has a sub-schema with different items of which some have "secret: true" attribute. Same goes for array.

I assume in such scenarios, if top level "secret: true" attribute is present, this top level attribute should have a precedence (and we could even ignore / skip iterating over child object / array attributes).

File: st2common/st2common/util/secrets.py [hunk 2]
```diff
@@ -66,6 +66,17 @@ def get_secret_parameters(parameters):
...
+    # If the parameter itself is secret, then skip all processing below it
+    # and return the type of this parameter.
+    #
+    # **This causes the _full_ object / array tree to be secret (no children will be shown).**
+    #
+    # **Important** that we do this check first, so in case this parameter
+    # is an `object` or `array`, and the user wants the full thing
+    # to be secret, that it is marked as secret.
+    if parameters.get('secret', False):
+        return parameters_type
```
Comment (Kami) [inline st2common/st2common/util/secrets.py]: Great :+1: 

That's also a more common scenario and it's much more performant :)

**Comment by Kami** [comment]
Thanks for good work on the tests :+1:
